objects/src/accounts/account_id.rs

```diff
@@ -140,14 +144,58 @@ impl AccountId {
         ];
         let mut current_digest = Self::compute_digest(current_seed, code_root, storage_root);
 
+        #[cfg(feature = "log")]
+        {
+            log::info!(
+                "Generating new account seed [pow={}, current_digest={}, current_seed={}]",
+                digest_pow(current_digest),
+                digest_hex(current_digest),
+                word_hex(current_seed),
+            );
+        }
+
+        #[cfg(feature = "log")]
+        let mut count = 0;
+        #[cfg(feature = "log")]
+        let mut best: (u32, Digest, Word) =
+            (digest_pow(current_digest), current_digest, current_seed);
+
         // loop until we have a seed that satisfies the specified account type.
         loop {
+            #[cfg(feature = "log")]
+            {
+                count += 1;
+                let current = (digest_pow(current_digest), current_digest, current_seed);
+                if best.0 <= current.0 {
+                    best = current;
+                }
+
+                if count % 500_000 == 0 {
+                    log::debug!(
+                        "Account seed loop [count={}, best_pow={}, best_digest={}, best_seed={}]",
+                        count,
+                        best.0,
+                        digest_hex(best.1),
+                        word_hex(best.2),
+                    );
+                }
+            }
+
             // check if the seed satisfies the specified account type
             if AccountId::validate_seed_digest(&current_digest).is_ok() {
                 if let Ok(account_id) = AccountId::try_from(current_digest[0]) {
                     if account_id.account_type() == account_type
                         && account_id.is_on_chain() == on_chain
                     {
+                        #[cfg(feature = "log")]
+                        {
+                            log::info!(
+                                "Found account seed [pow={}, current_digest={}, current_seed={}]",
+                                digest_pow(current_digest),
+                                digest_hex(current_digest),
+                                word_hex(current_seed),
+                            );
+                        }
```

Comment:
minor nit: I don't think extra curly braces are needed here since we are applying the feature to a single statement. I think the following should work:
```Rust
#[cfg(feature = "log")]
log::info!(
    "Found account seed [pow={}, current_digest={}, current_seed={}]",
    digest_pow(current_digest),
    digest_hex(current_digest),
    word_hex(current_seed),
);
```

Reply:
removed